e2e/fixtures/factory.go

```diff
@@ -51,25 +52,41 @@ const (
 
 // Factory is a helper struct to organize tests.
 type Factory struct {
-	*singleton
-	shutdownHooks          ShutdownHooks
-	chaosExperiments       []ChaosMeshExperiment
-	invariantShutdownHooks ShutdownHooks
-	beforeVersion          string
-	shutdownInProgress     bool
-	options                *FactoryOptions
+	shutdownHooks           ShutdownHooks
+	chaosExperiments        []ChaosMeshExperiment
+	invariantShutdownHooks  ShutdownHooks
+	shutdownInProgress      bool
+	beforeVersion           string
+	namespace               string
+	userName                string
+	options                 *FactoryOptions
+	randomGenerator         *rand.Rand
+	certificate             *corev1.Secret
+	namespaces              []string
+	controllerRuntimeClient client.Client
+	kubernetesClient        *kubernetes.Clientset
+	config                  *rest.Config
+	fdbVersion              fdbv1beta2.Version
 }
 
 // CreateFactory will create a factory based on the provided options.
 func CreateFactory(options *FactoryOptions) *Factory {
-	singleton, err := getSingleton(options)
+	configuration, err := getSingleton(options)
 	gomega.Expect(err).NotTo(gomega.HaveOccurred())
 
+	seed := time.Now().Unix()
```

Comment:
In theory we could add an option here to make use of the Ginkgo suite seed but I don't think there is a huge value as those e2e tests also depend on a lot of external things like the Kubernetes cluster and chaos-mesh.